packages/styled-system/src/utils/parse-gradient.ts

```diff
@@ -0,0 +1,72 @@
+import { Dict, memoizedGet as get } from "@chakra-ui/utils"
+import { ConfigStyle } from "@styled-system/core"
+
+const directionMap = {
+  "to-t": "to top",
+  "to-tr": "to top right",
+  "to-r": "to right",
+  "to-br": "to bottom right",
+  "to-b": "to bottom",
+  "to-bl": "to bottom left",
+  "to-l": "to left",
+  "to-tl": "to top left",
+}
+
+export enum ErrorMessages {
+  INVALID_SYNTAX = "Invalid value passed to prop: `bgGradient`",
+  INVALID_DIRECTION = "Direction is not valid. Defaulting to `top right`",
+}
+
+const DEFAULT_DIR: keyof typeof directionMap = "to-r"
+export const DEFAULT_TO_COLOR = "rgba(255, 0, 0, 0)"
+
+const globals = [
+  "none",
+  "-moz-initial",
+  "inherit",
+  "initial",
+  "revert",
+  "unset",
+]
+
+export function parseGradient(value: string, theme: Dict) {
+  if (globals.includes(value)) {
+    return value
+  }
+
+  const valueSplit = value.split(" ").filter(Boolean)
+
+  if (valueSplit.length > 2) {
+    throw Error(ErrorMessages.INVALID_SYNTAX)
+  }
+
+  let [dir, colors] = valueSplit
+
+  if (valueSplit.length === 1) {
+    colors = valueSplit[0]
+    dir = DEFAULT_DIR
+  }
+
+  let direction = directionMap[dir]
+
+  if (!direction) {
+    console.warn(ErrorMessages.INVALID_DIRECTION)
+    direction = directionMap[DEFAULT_DIR]
+  }
+
+  const colorsArr = colors
+    .split(":")
+    .filter(Boolean)
+    .map((color) => get(theme, `colors.${color}`, color))
+
+  if (colorsArr.length === 1) {
+    colorsArr.push(DEFAULT_TO_COLOR)
+  }
+
+  const colorsString = colorsArr.join(", ")
+
+  return `linear-gradient(${direction}, ${colorsString})`
+}
```

Comment:
I really like the simplicity of this, but I think it is not enough for real world use cases.
This implementation is limited to linear gradients and to a single gradient. 

Most gradients I needed to implement consisted of multiple combined gradients.

We could simplify the parsing algorithm by just replacing the color values:

```ts
export function parseGradient(value: string, theme: Dict) {
  return value.replace(/[a-zA-Z0-9_.]+/g, (substring) => {
    const color = get(theme, `colors.${substring}`, substring)
    return typeof color === "string" ? color : substring
  })
}
```

```ts
parseGradient("linear-gradient(pink.300, blue.500)", theme)
// browser -> linear-gradient(#F687B3, #3182ce)

parseGradient("radial-gradient(circle at center, red 0, blue, green 100%)", theme)
// browser -> radial-gradient(circle at center, #f00 0, #00f, #008000 100%)
```

With this we would be able to support `linear-gradient()`, `radial-gradient()`, `repeating-linear-gradient()`, `repeating-radial-gradient()` etc. And chakra users won't have to learn a new gradient syntax.

This idea came to me when playing around with the new syntax. What do you think about this approach?

Reply:
I like where this is going, Tim. Thanks for taking the time.

- The constraints to linear-gradient is rather intentional considering it's a common use-case. I didn't think about supporting other types of gradients. Can you show me a real-world use-case based on your project?
- This doesn't mean the user cannot use the typical `backgroundImage` prop to add more complex gradients. I'd rather say for simple linear gradients, you can use this prop, else use the `backgroundImage` or `background` prop
- I'd love to keep your simplification though coz it reduces the LOC.

Let me think on this a bit more